### ISSUE TITLE
设备内存增长似乎不符合预期

### ISSUE THREAD

**Issue by LeoLau94** [BD-Seed-HHW/xpu_graph#324]
### Describe the bug🐛

目前我们有分桶场景需要支持，由于分桶capture场景中是从最大桶开始的，在默认行为下，除了第一次，每次分桶都不应该预期之外的设备内存增长行为，但实际并不是这样的，这不符合预期。

### Environment details

PyTorch: 2.5.1
XpuGraph: 0.2.0

**Comment by LeoLau94** [comment]
似乎类似arena allocator的设计导致的
https://en.wikipedia.org/wiki/Region-based_memory_management

**Comment by LeoLau94** [comment]
已知信息是acl graph不支持多图复用

**Comment by LeoLau94** [comment]
https://docs.pytorch.org/docs/stable/notes/cuda.html#graph-memory-management

**Comment by wbigat** [comment]
我们正在提供一个config给用户，方便用户指定某次compile model过程中出现的graph 使用传入的内存池。

因此当多张图想要复用内存池时，可以在config 初始化时指定相同的pool即可

大致的使用方式是：
```
import torch_npu
reused_pool = _pool = torch.npu.graph_pool_handle()

import torchair
config = torchair.CompilerConfig()
config.aclgraph_config.use_custom_pool = reused_pool
npu_backend = torchair.get_npu_backend(compiler_config=config)
```

**Comment by LeoLau94** [comment]
<img width="908" height="208" alt="Image" src="https://github.com/user-attachments/assets/24a6b821-02a9-45a9-908e-f96ed96c68a8" />
当前的输出不是复用的

出现增长为0的情况，可能是allcoator向system申请的时候，按照一次策略多申请了，而后续桶越分越小，所以有部分桶正好没有触发到extend mem的水线
